libs/community/langchain_community/chat_models/notdiamond.py

```diff
@@ -0,0 +1,416 @@
+        **kwargs: Any,
+    ) -> ChatResult:
+        message_dicts, params = self._create_message_dicts(messages, stop)
+        should_stream = stream if stream is not None else self.streaming
+
+        if should_stream:
+            stream_iter = self._astream(
+                messages=messages, stop=stop, run_manager=run_manager, **kwargs
+            )
+            return await agenerate_from_stream(stream_iter)
+
+        params = {**params, **kwargs}
+        
+        response, session_id, provider = await acompletion_with_retry(
+            self,
+            messages=message_dicts,
+            run_manager=run_manager, 
+            **params
+        )
+        result = {
+            'response': response,
+            'session_id': session_id,
+            'provider': provider
+        }
+        return self._create_chat_result(result)
+
+    def _stream(
+        self,
+        messages: List[BaseMessage],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> Iterator[ChatGenerationChunk]:
+        message_dicts, params = self._create_message_dicts(messages, stop)
+        params = {**params, **kwargs, "stream": True}
+
+        generator_response = completion_with_retry(
+            self,
+            messages=message_dicts,
+            run_manager=run_manager,
+            **params,
+        )
+        for chunk in generator_response:
+            if not chunk.content:
+                continue
+            cg_chunk = ChatGenerationChunk(message=chunk)
+            if run_manager:
+                run_manager.on_llm_new_token(chunk.content, chunk=cg_chunk)
+            yield cg_chunk
+
+    async def _astream(
+        self,
+        messages: List[BaseMessage],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> AsyncIterator[ChatGenerationChunk]:
+        message_dicts, params = self._create_message_dicts(messages, stop)
+        params = {**params, **kwargs, "stream": True}
+
+        generator_response = completion_with_retry(
+            self,
+            messages=message_dicts,
+            run_manager=run_manager,
+            **params,
+        )
+        async for chunk in await generator_response:
+            if not chunk.content:
+                continue
+            cg_chunk = ChatGenerationChunk(message=chunk)
+            if run_manager:
+                await run_manager.on_llm_new_token(chunk.content, chunk=cg_chunk)
+            yield cg_chunk
+
+    def _create_chat_result(self, response: Mapping[str, Any]) -> ChatResult:
+        res = response['response']
+        gen = ChatGeneration(
+            message=AIMessage(content=res.content),
+            generation_info=dict(finish_reason=res.response_metadata.get("finish_reason")),
+        )
+        input_tokens = res.usage_metadata.get("input_tokens", {})
```

Comment:
Might be better to default to `None` rather than empty dict but it's best to maintain the same behaviour as other integrations